src/main/java/org/jenkinsci/remoting/protocol/IOHub.java

```diff
@@ -83,6 +92,10 @@
      * Our selector.
      */
     private final Selector selector;
+    private volatile boolean winWatcherSelectorIsWaiting = false;
+    private volatile boolean ioHubRunning = false;
+    private long winWatcherSelectorPollTimestamp = 0;
```

Comment:
you can make this much simpler by removing this and just doing an unconditional wakeup() every second.
the worst that happens if you have 2 wakeups in an exceptionally short time (one for data then immediatly a spurious wakeup) - but that seems reasonable.

Reply:
OK, will do. It does not cause performance degradation, but I agree it is a bit overcomplicated